content/2015.12.loadtesting.with.oauth.rst

```diff
@@ -0,0 +1,214 @@
+                         oauth_server_url=oauth.DEFAULT_SERVER_URL):
+    
+        if scopes is None:
+            scopes = ['profile']
+    
+        client = core.Client(server_url=account_server_url)
+        session = client.login(email, password)
+    
+        url = urlparse(oauth_server_url)
+        audience = "%s://%s/" % (url.scheme, url.netloc)
+    
+        bid_assertion = session.get_identity_assertion(audience)
+        oauth_client = oauth.Client(server_url=oauth_server_url)
+        token = oauth_client.authorize_token(bid_assertion,
+                                             ' '.join(scopes),
+                                             client_id)
+        return token
+
+    EMAIL = 'your.account@email-address.com'
+    PASSWORD = 'Y0UR 4CC0UNT P4SSW0RD'
+	token = get_bearer_token(EMAIL, PASSWORD)
+    print('export OAUTH_BEARER_TOKEN="%s"' % token)
+
+
+Generating a token for a new account
+------------------------------------
+
+You may want to create accounts on the stage environment and generate
+tokens for them.
+
+.. code-block:: python
+
+    from uuid import uuid4
+    from fxa.core import Client
+
+    FXA_SERVER_URL = https://api-accounts.stage.mozaws.net/v1
+    PASSWORD = uuid4()
+	EMAIL = "project-name-%s@restmail.net" % PASSWORD
+
+    # Constants
+    FXA_ERROR_ACCOUNT_EXISTS = 101
+
+    if 'stage' not in FXA_SERVER_URL:
+        raise Exception("Please use the FxA stage server.")
+
+    client = Client(FXA_SERVER_URL)
+
+    try:
+        client.create_account(EMAIL,
+                              password=PASSWORD,
+                              preVerified=True)  # This only works on stage
+    except errors.ClientError as e:
+        if e.errno != FXA_ERROR_ACCOUNT_EXISTS:
+            raise
+
+	token = get_bearer_token(EMAIL, PASSWORD)
+    print('export OAUTH_BEARER_TOKEN="%s"' % token)
+
+
+Working with multiple Bearer Tokens
+===================================
+
+If you need to use multiple user, you can add other environment
+variable.
+
+You can also set multiple users Bearer tokens separated by comma in
+your ENV variable and then split on it to choose one randomly in your
+loadtest:
+
+.. code-block:: bash
+
+    export OAUTH_BEARER_TOKEN="
+        dd7e6b6d0f4d44d0df5d6b73afdfbad41c48c4abdaba50e428ce070f9d4c75b5,
+        b6af04a44aa0f5a6b3a3affbaa41c48c4abaaba50e428ce030f9a4cb356aa36e,
+        a44aa0f5a6b3a3affbaa41c1c48c4abaaba50e428ce030f9428ce070f9d4c75b
+        "
+
+You can also create multiple environment variables:
+
+.. code-block:: bash
+
+    export OAUTH_BEARER_TOKEN_SCOPE_PROFILE="dd7e6b6d0f4d44d0df5d6b73afdfbad41c48c4abdaba50e428ce070f9d4c75b5"
+    export OAUTH_BEARER_TOKEN_SCOPE_KINTO="
+        b6af04a44aa0f5a6b3a3affbaa41c48c4abaaba50e428ce030f9a4cb356aa36e,
+        a44aa0f5a6b3a3affbaa41c1c48c4abaaba50e428ce030f9428ce070f9d4c75b
+        "
+
+Conclusion
+==========
+
+After thinking this through, it doesn't look so complex to loadtest a
```

Comment:
I hope that after reading this article, you aren't afraid anymore of loadtesting an OAuth-based service!

Tak-aways: